### DISCUSSION TITLE
Support for openmaptiles

### DISCUSSION THREAD

**Discussion by nyurik** [maplibre/martin#250] [Ideas]
The [openmaptiles](https://github.com/openmaptiles/openmaptiles) project also supports postgis-based tile generation - it can generate giant SQL statement (as either prepared statement, query, or function creation) that produces MVT from z,x,y.   Would it be possible to add support for the openmaptiles layer format in martin as well?  See [the postgis server](https://github.com/openmaptiles/openmaptiles-tools#test-tiles) (simple test server for testing purposes)

**Comment by stepankuzmin** [comment]
Hi @nyurik! That's a great idea! Unfortunately, I don't have enough bandwidth for implementing this now. But I'll keep it in mind and maybe implement this in the future releases.

**Comment by littlebtc** [comment]
After trying OpenMapTiles, I found that the docker-compose file from https://github.com/openmaptiles/openmaptiles/ still using old PostGIS with old `ST_AsMVT` syntax. A major PostGIS update for OpenMapTiles is needed in order to run it :(

**Comment by nyurik** [comment]
@littlebtc most of the tools have already been updated to the latest. https://github.com/openmaptiles/openmaptiles-tools has been receiving tons of changes in the last weeks (just today I made a number of changes to postserve -- a simple debug-only version of `martin` that uses OpenMapTile layer defenition)

I have been doing some of it as part of https://github.com/sophox/openmaptiles too, and very soon I plan to have a working docker image with PostgreSQL 12 + PostGIS 3.0 (they are massively faster for MVTs, and were released a week ago).   My suggestion - for testing, simply swap the order of the parameters to ST_AsMVT -- and your code will work right away with the existing quick start. I will have the quickstart ready within the next few days to run on the latest platform.

Thanks!

**Reply by JanikoNaber** [reply]
I know, this is long time ago. But do you know if there is now the possibility to run openmaptiles with Martin?

**Comment by littlebtc** [comment]
@nyurik Great! I will give it a try if I have time.

**Comment by jike212** [comment]
Any updates on this? I'd love to see support for OpenMapTiles schema in Martin.

**Comment by vinayakkulkarni** [comment]
Is there a script/utility that can provide this?

**Comment by JanikoNaber** [comment]
Hi all,
now a few years later, I still like the idea from @nyurik. Does anyone know if its somehow possible to serve generated openmaptiles vector tiles with Martin?
I think that must not necessarily be done on the fly, like the test server from openmaptiles. I'm not deep enough into the topic, but shouldn't it be possible to get a PostGIS database form openmaptiles instead of the mbtiles files, and serve this database via Martin?
Thank you for your feedback.

Best 
Janiko

**Reply by nyurik** [reply]
I haven't tried it, but I don't see why Martin can't use the generated omt sql to get tiles

**Reply by vinayakkulkarni** [reply]
Would be an really interesting PoC @prashis?

**Comment by nyurik** [comment]
**update** turns out the current implementation of Martin would not be directly compatible with OpenMapTiles because martin expect some query params that will be passed to SQL. So the function that only has `z,x,y`, or a function that uses `zoom` instead of `z` it wouldn't work.  I made a proposal of how we can enhance Martin to support such cases in #378
